platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/EventMetadata.java

```diff
@@ -0,0 +1,273 @@
+        } else {
+            // this event does not have a self parent
+            this.selfParent = null;
+            this.otherParents = allParents;
+        }
+        this.generation = calculateGeneration(allParents);
+        this.timeCreated = HapiUtils.asInstant(
+                Objects.requireNonNull(gossipEvent.eventCore().timeCreated(), "The timeCreated must not be null"));
+        this.transactions =
+                Objects.requireNonNull(gossipEvent.eventTransaction(), "transactions must not be null").stream()
+                        .map(TransactionWrapper::new)
+                        .toList();
+    }
+
+    private static long calculateGeneration(@NonNull final List<EventDescriptorWrapper> allParents) {
+        return 1
+                + Objects.requireNonNull(allParents).stream()
+                        .mapToLong(d -> d.eventDescriptor().generation())
+                        .max()
+                        .orElse(EventConstants.GENERATION_UNDEFINED);
+    }
+
+    /**
+     * Returns the software version of the node that created this event.
+     *
+     * @return the software version of the node that created this event
+     */
+    @NonNull
+    public SoftwareVersion getSoftwareVersion() {
+        return softwareVersion;
+    }
+
+    /**
+     * The ID of the node that created this event.
+     *
+     * @return the ID of the node that created this event
+     */
+    @NonNull
+    public NodeId getCreatorId() {
+        return creatorId;
+    }
+
+    /**
+     * Get the event descriptor for the self parent.
+     *
+     * @return the event descriptor for the self parent
+     */
+    @Nullable
+    public EventDescriptorWrapper getSelfParent() {
+        return selfParent;
+    }
+
+    /**
+     * Get the hash of the self parent.
+     *
+     * @return the hash of the self parent
+     */
+    @Nullable
+    public Hash getSelfParentHash() {
+        if (selfParent == null) {
+            return null;
+        }
+        return selfParent.hash();
+    }
+
+    /**
+     * Get the event descriptors for the other parents.
+     *
+     * @return the event descriptors for the other parents
+     */
+    @NonNull
+    public List<EventDescriptorWrapper> getOtherParents() {
+        return otherParents;
+    }
+
+    /**
+     * @return the hash of the other parent
+     */
+    @Nullable
+    @Deprecated
+    public Hash getOtherParentHash() {
+        if (otherParents == null || otherParents.isEmpty()) {
+            return null;
+        }
+        if (otherParents.size() == 1) {
+            return otherParents.getFirst().hash();
+        }
+        // 0.46.0 adds support for multiple other parents in the serialization scheme, but not yet in the
+        // implementation. This exception should never be reached unless we have multiple parents and need to
+        // update the implementation.
+        throw new UnsupportedOperationException("Multiple other parents is not supported yet");
```

Comment:
Probably should add a unit test to verify this branch throws an exception and that this code isn't deleted by accident before we add multiple other parents intentionally.

Reply:
i removed the whole deprecated method